modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorage.java

```diff
@@ -226,22 +262,37 @@ public void committedGroupConfiguration(RaftGroupConfiguration config) {
         assert checkpointTimeoutLock.checkpointLockIsHeldByThread();
 
         CheckpointProgress lastCheckpoint = checkpointManager.lastCheckpointProgress();
-
         UUID lastCheckpointId = lastCheckpoint == null ? null : lastCheckpoint.id();
 
-        meta.lastGroupConfig(lastCheckpointId, groupConfigToBytes(config));
+        byte[] raftGroupConfigBytes = raftGroupConfigToBytes(config);
+
+        raftGroupConfigReadWriteLock.writeLock().lock();
+
+        try {
+            if (meta.lastRaftGroupConfigFirstPageId() == BlobStorage.NO_PAGE_ID) {
+                long configPageId = blobStorage.addBlob(raftGroupConfigBytes);
+
+                meta.lastRaftGroupConfigFirstPageId(lastCheckpointId, configPageId);
+            } else {
+                blobStorage.updateBlob(meta.lastRaftGroupConfigFirstPageId(), raftGroupConfigBytes);
+            }
+        } catch (IgniteInternalCheckedException e) {
+            throw new StorageException("Cannot save committed group configuration, groupId=" + groupId + ", partitionId=" + groupId, e);
+        } finally {
+            raftGroupConfigReadWriteLock.writeLock().unlock();
+        }
     }
 
     @Nullable
-    private static RaftGroupConfiguration groupConfigFromBytes(byte @Nullable [] bytes) {
+    private static RaftGroupConfiguration raftGroupConfigFromBytes(byte @Nullable [] bytes) {
```

Comment:
By the way, why do we have to mention RAFT here? There's a chance of us having more replication protocol in the future, can we try and abstract ourselves a little bit? At least stop renaming old methods into a new that mention RAFT even more.

It would be a good idea to rename RaftGroupConfiguration as well. Or even better. I don't understand why it's a storages' task to convert an object into a byte array. Looks like a wrong design decision.

Reply:
I added a ticket https://issues.apache.org/jira/browse/IGNITE-18405 because renaming this class is out of scope of this PR.

Reply:
Also added https://issues.apache.org/jira/browse/IGNITE-18408 about storing bytes, not config object

Reply:
Thank you!

Reply:
Renamed internal fields/constants/methods (not relating to the API)